src/components/GoalsComponents/AddGoal/AddGoalForm.tsx

```diff
@@ -42,15 +42,22 @@ export const AddGoalForm: React.FC<AddGoalFormProps> = ({ goalId, setShowAddGoal
       [e.target.name]: value,
     });
   };
-  const freqRegex = /[D|d]+aily|AILY/;
-  const freq = formInputData.inputGoal.match(freqRegex);
+  const daily = /daily/i;
+  const once = /once/i;
+  const weekly = /weekly/i;
+  const freqDaily = formInputData.inputGoal.match(daily);
```

Comment:
can't we  just lowercase all entire inputGoal while checking. this way we can dodge the case-sensitive issue

Reply:
The /i approach has the advantage that it is simple. It might be harder to read for a dev not familiar with the regex options so a comment would be good.

I believe /i doesn't remove the formatting for the matched text? Anyway that is not important for this as we choose/hardcode the case of the property title in the code.

Reply:
If the issue I mentioned can be reproduced and it is not clear why the regex fails then Tushar's approach is a good workaround to not spend too much time on this.

Reply:
Actually, I think we should go with Tushar's approach.